### PR TITLE
feat(secret-manager): add tutorial on how to use scaleway secret mana…

### PR DESCRIPTION
**Pull request by dimq**
…ger in kapsule

### Your checklist for this pull request

- [x] Check that the commit messages match our requested structure.
- [x] Name your pull request according to the [contribution guidelines](https://github.com/scaleway/docs-content/blob/main/docs/CONTRIBUTING.md).

### Description
Tutorial on how to use Scaleway secret manager on Kubernetes Kapsule.

### CODE REVIEW

File: tutorials/external-secrets/index.mdx [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+  - You have [configured your SSH Key](/console/my-project/how-to/create-ssh-key/)
+  - You have [created a Kapsule cluster](/containers/kubernetes/how-to/create-cluster/)
+  - [kubectl](/containers/kubernetes/how-to/connect-cluster-kubectl/) is working on your machine
+  - `helm`, the Kubernetes [package manager](https://helm.sh/), is installed on your local machine (version 3.2+)
+</Message>
+
+## Preparing the Kubernetes Kapsule Cluster
+
+1. Ensure you are connected to your cluster and that `kubectl` and `helm` are installed on your local machine.
+2. Add the External Secrets Helm repo update it:
```
Comment (azert9) [inline tutorials/external-secrets/index.mdx:38]: Il doit manquer un "and"

File: tutorials/external-secrets/index.mdx [hunk 2]
```diff
@@ -0,0 +1,127 @@
+---
+meta:
+  title: Deploying External secrets on Kubernetes Kapsule
```
Comment (nedaehm) [inline tutorials/external-secrets/index.mdx:3]:
```suggestion
  title: Deploying External Secrets on Kubernetes Kapsule
```

File: tutorials/external-secrets/index.mdx [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+  description: Learn how to deploy External Secret on Kubernetes Kapsule integrated with Scaleway Secret Manager
```
Comment (nedaehm) [inline tutorials/external-secrets/index.mdx:4]:
```suggestion
  description: Learn how to deploy External Secrets on Kubernetes Kapsule integrated with Scaleway Secret Manager
```

File: tutorials/external-secrets/index.mdx [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+content:
+  h1: Deploying External secrets on Kubernetes Kapsule
```
Comment (nedaehm) [inline tutorials/external-secrets/index.mdx:6]:
```suggestion
  h1: Deploying External Secrets on Kubernetes Kapsule
```

File: tutorials/external-secrets/index.mdx [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+spec:
+    refreshInterval: 20s
+    secretStoreRef:
+        kind: SecretStore
+        name: secret-store
+    data:
+      - secretKey: password  # key in the kubernetes secret
+        remoteRef:
+          key: id:<SECRET_ID in the secret store>
+          version: latest
```
Comment (azert9) [inline tutorials/external-secrets/index.mdx:108]: Il faudrait peut être donner latest_enabled en exemple ?

File: tutorials/external-secrets/index.mdx [hunk 6]
```diff
@@ -0,0 +1,127 @@
+---
+meta:
+  title: Deploying External secrets on Kubernetes Kapsule
+  description: Learn how to deploy External Secret on Kubernetes Kapsule integrated with Scaleway Secret Manager
+content:
+  h1: Deploying External secrets on Kubernetes Kapsule
+  paragraph: Learn how to deploy External secrets on Kubernetes Kapsule integrated with Scaleway Secret Manager
```
Comment (nedaehm) [inline tutorials/external-secrets/index.mdx:7]:
```suggestion
  paragraph: Learn how to deploy External Secrets on Kubernetes Kapsule integrated with Scaleway Secret Manager
```

File: tutorials/external-secrets/index.mdx [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+tags: Kapsule cluster compute Kubernetes External secrets Scaleway secret manager
```
Comment (nedaehm) [inline tutorials/external-secrets/index.mdx:8]:
```suggestion
tags: kapsule-cluster kubernetes external-secrets secret-management
```

File: tutorials/external-secrets/index.mdx [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+          key: id:<SECRET_ID in the secret store>
+          version: latest
+```
+
+## Uninstalling
+
+Before continuing, ensure that all external-secret resources that have been created by users have been deleted. You can check for any existing resources with the following command:
+
+```
+$ kubectl get SecretStores,ClusterSecretStores,ExternalSecrets --all-namespaces
```
Comment (azert9) [inline tutorials/external-secrets/index.mdx:116]: Il y a d'autres types de resources (d'après https://external-secrets.io/v0.7.2/api/components/):
- ClusterExternalSecret
- PushSecret

File: tutorials/external-secrets/index.mdx [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+  h1: Deploying External secrets on Kubernetes Kapsule
+  paragraph: Learn how to deploy External secrets on Kubernetes Kapsule integrated with Scaleway Secret Manager
+tags: Kapsule cluster compute Kubernetes External secrets Scaleway secret manager
+categories:
+  - containers
+  - kubernetes
+  - secrets
+  - security
+dates:
+  validation: 2023-17-02
```
Comment (nedaehm) [inline tutorials/external-secrets/index.mdx:15]:
```suggestion
  validation: 2023-02-21
```

File: tutorials/external-secrets/index.mdx [hunk 10]
```diff
@@ -0,0 +1,127 @@
...
+  posted: 2022-17-02
```
Comment (nedaehm) [inline tutorials/external-secrets/index.mdx:16]:
```suggestion
  posted: 2023-02-21
```

File: tutorials/external-secrets/index.mdx [hunk 11]
```diff
@@ -0,0 +1,127 @@
...
+---
+
+## External Secrets - Overview
+
+[External Secrets](https://external-secrets.io) Is a Kubernetes operator allowing you to handle the lifecycle of secrets from external providers.
```
Comment (nedaehm) [inline tutorials/external-secrets/index.mdx:21]:
```suggestion
[External Secrets](https://external-secrets.io) is a Kubernetes operator that allows you to manage the lifecycle of your secrets from external providers.
```

File: tutorials/external-secrets/index.mdx [hunk 12]
```diff
@@ -0,0 +1,127 @@
...
+In this tutorial you will learn how to deploy External Secrets and its services on [Kubernetes Kapsule](/containers/kubernetes/concepts/#kubernetes-kapsule), the managed Kubernetes service from Scaleway.
+
+<Macro id="iam-requirements" />
+
+<Message type="requirement">
+  - You have an account and are logged into the [Scaleway console](https://console.scaleway.com)
+  - You have [configured your SSH Key](/console/my-project/how-to/create-ssh-key/)
+  - You have [created a Kapsule cluster](/containers/kubernetes/how-to/create-cluster/)
+  - [kubectl](/containers/kubernetes/how-to/connect-cluster-kubectl/) is working on your machine
+  - `helm`, the Kubernetes [package manager](https://helm.sh/), is installed on your local machine (version 3.2+)
```
Comment (nedaehm) [inline tutorials/external-secrets/index.mdx:32]:
```suggestion
  - You have installed `helm`, the Kubernetes [package manager](https://helm.sh/), on your local machine (version 3.2 or latest)
```

File: tutorials/external-secrets/index.mdx [hunk 13]
```diff
@@ -0,0 +1,127 @@
...
+
+In this tutorial you will learn how to deploy External Secrets and its services on [Kubernetes Kapsule](/containers/kubernetes/concepts/#kubernetes-kapsule), the managed Kubernetes service from Scaleway.
+
+<Macro id="iam-requirements" />
+
+<Message type="requirement">
+  - You have an account and are logged into the [Scaleway console](https://console.scaleway.com)
+  - You have [configured your SSH Key](/console/my-project/how-to/create-ssh-key/)
+  - You have [created a Kapsule cluster](/containers/kubernetes/how-to/create-cluster/)
+  - [kubectl](/containers/kubernetes/how-to/connect-cluster-kubectl/) is working on your machine
```
Comment (nedaehm) [inline tutorials/external-secrets/index.mdx:31]:
```suggestion
  - You have configured [kubectl](/containers/kubernetes/how-to/connect-cluster-kubectl/)
```

File: tutorials/external-secrets/index.mdx [hunk 14]
```diff
@@ -0,0 +1,127 @@
...
+  - `helm`, the Kubernetes [package manager](https://helm.sh/), is installed on your local machine (version 3.2+)
+</Message>
+
+## Preparing the Kubernetes Kapsule Cluster
+
+1. Ensure you are connected to your cluster and that `kubectl` and `helm` are installed on your local machine.
```
Comment (nedaehm) [inline tutorials/external-secrets/index.mdx:37]:
```suggestion
1. Make sure you are connected to your cluster and that `kubectl` and `helm` are installed on your local machine.
```

File: tutorials/external-secrets/index.mdx (hunk 1 above)
Comment (nedaehm) [inline tutorials/external-secrets/index.mdx]:
```suggestion
2. Add the External Secrets helm repo and update it using the following command:
```

File: tutorials/external-secrets/index.mdx [hunk 15]
```diff
@@ -0,0 +1,127 @@
...
+2. Add the External Secrets Helm repo update it:
+    ```
+    $ helm repo add external-secrets https://charts.external-secrets.io
+    $ helm repo update
+    ```
+
+## Deploying External Secrets
+
+To automatically install and manage the CRDs as part of your Helm release, you must add the --set installCRDs=true flag to your Helm installation command. Uncomment the relevant line in the next steps to enable this.
```
Comment (nedaehm) [inline tutorials/external-secrets/index.mdx]:
```suggestion
To automatically install and manage the CRDs as part of your Helm release, you must add the `--set installCRDs=true` flag to your Helm installation command. Uncomment the relevant line in the next steps to enable this.
```

File: tutorials/external-secrets/index.mdx [hunk 16]
```diff
@@ -0,0 +1,127 @@
...
+```
+$ helm install external-secrets \
+   external-secrets/external-secrets \
+    -n external-secrets \
+    --create-namespace \
+    --set image.repository=rg.fr-par.scw.cloud/external-secrets/external-secrets \
+    --set image.tag=v1
+  # --set installCRDs=true
+```
+## Create a secret containing your SCW credentials
```
Comment (nedaehm) [inline tutorials/external-secrets/index.mdx:65]:
```suggestion
## Create a secret containing your Scaleway's API key information
```

File: tutorials/external-secrets/index.mdx [hunk 17]
```diff
@@ -0,0 +1,127 @@
...
+
+```
+$ echo -n 'KEYID' > ./access-key
+$ echo -n 'SECRETKEY' > ./secret-access-key
+$ kubectl create secret generic scwsm-secret --from-file=./access-key --from-file=./secret-access-key
+```
+## Create your first SecretStore
+
+Secret Manager is a regionalized product so you'll need to create a secret store by region.
```
Comment (nedaehm) [inline tutorials/external-secrets/index.mdx:74]:
```suggestion
Secret Manager is a regionalized product so you will need to create a secret store by region.
```

File: tutorials/external-secrets/index.mdx [hunk 18]
```diff
@@ -0,0 +1,127 @@
...
+    data:
+      - secretKey: password  # key in the kubernetes secret
+        remoteRef:
+          key: id:<SECRET_ID in the secret store>
+          version: latest
+```
+
+## Uninstalling
+
+Before continuing, ensure that all external-secret resources that have been created by users have been deleted. You can check for any existing resources with the following command:
```
Comment (nedaehm) [inline tutorials/external-secrets/index.mdx:121]:
```suggestion
Make sure you have deleted all external-secret resources you might have created beforehand. You can check for any existing resources with the following command:
```

File: tutorials/external-secrets/index.mdx [hunk 19]
```diff
@@ -0,0 +1,135 @@
...
+1. Make sure you are connected to your cluster and that `kubectl` and `helm` are installed on your local machine.
+2. Add the External Secrets Helm repo and update it:
+    ```
+    $ helm repo add external-secrets https://charts.external-secrets.io
+    $ helm repo update
+    ```
+
+## Deploying External Secrets
+```
+$ helm upgrade --install external-secrets  external-secrets/external-secrets
```
Comment (azert9) [inline tutorials/external-secrets/index.mdx:46]: Il faudrait rajouter les antislash en fin de ligne pour faciliter le copier-coller

File: tutorials/external-secrets/index.mdx [hunk 20]
```diff
@@ -0,0 +1,135 @@
...
+    --set webhook.image.repository=rg.fr-par.scw.cloud/external-secrets/external-secrets
+    --set webhook.image.tag=v1
+```
+
+The CRD's can only be installed for now with the corresponding git branch:
+
+```
+$ git clone -b feature/scaleway-provider git@github.com:azert9/external-secrets.git
+$ cd external-secrets
+$ make crds-install
```
Comment (azert9) [inline tutorials/external-secrets/index.mdx:62]: `crds-install` -> `crds.install`

File: tutorials/external-secrets/index.mdx [hunk 21]
```diff
@@ -0,0 +1,135 @@
...
+  title: Deploying External Secrets on Kubernetes Kapsule
+  description: Learn how to deploy External Secrets on Kubernetes Kapsule integrated with Scaleway Secret Manager
+content:
+  h1: Deploying External Secrets on Kubernetes Kapsule
+  paragraph: Learn how to deploy External Secrets on Kubernetes Kapsule integrated with Scaleway Secret Manager
+tags: kapsule-cluster kubernetes external-secrets secret-management
+categories:
+  - containers
+  - kubernetes
+  - secrets
```
Comment (nedaehm) [inline tutorials/external-secrets/index.mdx:12]:
```suggestion
```

File: tutorials/external-secrets/index.mdx [hunk 22]
```diff
@@ -0,0 +1,135 @@
...
+  - security
```
Comment (nedaehm) [inline tutorials/external-secrets/index.mdx:13]:
```suggestion
```

File: tutorials/external-secrets/index.mdx (hunk 21 above)
Comment (nedaehm) [reply tutorials/external-secrets/index.mdx:12]: we don't have this category :)

File: tutorials/external-secrets/index.mdx (hunk 22 above)
Comment (nedaehm) [reply tutorials/external-secrets/index.mdx:13]: same comment as above

File: tutorials/external-secrets/index.mdx [hunk 23]
```diff
@@ -0,0 +1,135 @@
...
+    --create-namespace \
+    --set image.repository=rg.fr-par.scw.cloud/external-secrets/external-secrets \
+    --set image.tag=v1 \
+    --set certController.image.repository=rg.fr-par.scw.cloud/external-secrets/external-secrets \
+    --set certController.image.tag=v1 \
+    --set webhook.image.repository=rg.fr-par.scw.cloud/external-secrets/external-secrets \
+    --set webhook.image.tag=v1
+```
+
+The CRD's can only be installed for now with the corresponding git branch:
```
Comment (RoRoJ) [inline tutorials/external-secrets/index.mdx:57]:
```suggestion
The CRDs can only be installed for now with the corresponding git branch:
```

File: tutorials/external-secrets/index.mdx [hunk 24]
```diff
@@ -0,0 +1,135 @@
...
+
+```
+$ git clone -b feature/scaleway-provider git@github.com:azert9/external-secrets.git
+$ cd external-secrets
+$ make crds.install
+```
+
+## Create a secret containing your Scaleway's API key information
```
Comment (RoRoJ) [inline tutorials/external-secrets/index.mdx:65]:
```suggestion
## Create a secret containing your Scaleway API key information
```

**Review by RoRoJ — APPROVED** [review]

**Review by nedaehm — APPROVED** [review]

**Pull request merged**